design/operator-framework.md

```diff
@@ -0,0 +1,18 @@
+# Moving Rook operators to Operator framework
+
+## Background
+
+Rook operators were developed with the help of in ground operator framework called operatorKit. The purpose of the operator kit was to provide a common library for implementing operators. Since the operator kit is not getting much development and with time a more advanced framework for developing operator called operator framework came into the market it would be a nice strategy to leverage the features offered by it.
+
+## Operator framework
```

Comment:
Per discussion in the community meeting, more details about the integration with Rook would be helpful so we understand what needs to change and what benefits the Rook project will gain from building on the operator sdk. 
- Clearly, replacing the rook operator kit is a good thing since it's not actively being improved
- Integration with the controller runtime is already planned as mentioned [here](https://github.com/rook/rook/issues/1981#issuecomment-524884768). 
- The operator sdk is already used in rook to generate the CSV, as seen [here](https://github.com/rook/rook/blob/master/cluster/olm/ceph/generate-rook-csv.sh)